Source/WebCore/rendering/MarkedText.cpp

```diff
@@ -290,7 +300,14 @@ Vector<MarkedText> MarkedText::collectForDocumentMarkers(const RenderText& rende
 #endif
         case DocumentMarker::Type::TextMatch: {
             auto [clampedStart, clampedEnd] = selectableRange.clamp(marker->startOffset(), marker->endOffset());
-            markedTexts.append({ clampedStart, clampedEnd, markedTextTypeForMarkerType(marker->type()), marker.get() });
+
+            auto markedTextType = markedTextTypeForMarkerType(marker->type());
+#if ENABLE(UNIFIED_TEXT_REPLACEMENT)
+            if (marker->type() == DocumentMarker::Type::UnifiedTextReplacement && std::get<DocumentMarker::UnifiedTextReplacementData>(marker->data()).applyIndicator == DocumentMarker::UnifiedTextReplacementData::ApplyIndicator::No)
+                markedTextType = MarkedText::Type::Unmarked;
+#endif
```

Comment:
I'm confused why this logic is under `DocumentMarker::Type::TextMatch`.

Reply:
because the logic is common to a bunch of different types, including `UnifiedTextReplacement`

Reply:
I should have expanded the context LOL